bottlecap/src/traces/stats_generator.rs

```diff
@@ -36,31 +35,7 @@ impl StatsGenerator {
                 // Generate stats for each span in the trace
                 for chunk in &trace.chunks {
                     for span in &chunk.spans {
-                        let stats = StatsEvent {
-                            time: span.start.try_into().unwrap_or_default(),
-                            aggregation_key: AggregationKey {
-                                env: span
-                                    .meta
-                                    .get("env")
-                                    .cloned()
-                                    .unwrap_or("unknown-env".to_string()),
-                                service: span.service.clone(),
-                                name: span.name.clone(),
-                                resource: span.resource.clone(),
-                                r#type: span.r#type.clone(),
-                                is_trace_root: span.parent_id == 0,
-                            },
-                            stats: Stats {
-                                hits: 1,
-                                errors: span.error,
-                                duration: span.duration,
-                                top_level_hits: span
-                                    .metrics
-                                    .get("_dd.top_level")
-                                    .map_or(0.0, |v| *v),
-                            },
-                        };
-                        if let Err(err) = self.stats_concentrator.add(stats) {
+                        if let Err(err) = self.stats_concentrator.add(span) {
```

Comment:
Just pass the `pb::Span`, which is what the concentrator takes in. No need to construct `StatsEvent`.